File: frontend/src/types/quiz.ts
export type QuizOption = {
  code: string;
  label_en: string;
  label_zh: string;
};

export type QuizQuestion = {
  code: string;
  question_en: string;
  question_zh: string;
  answer_type: "single" | "multiple";
  options: QuizOption[];
};

export type QuizSession = {
  session_token: string;
  language: "en" | "zh";
  status: string;
};

export type Recommendation = {
  product_id: string;
  product_code: string;
  product_name_en: string;
  product_name_zh: string | null;
  score: number;
  tier: "Excellent" | "Good" | "Potential" | "Do Not Recommend";
  reasons: string[];
};

export type RecommendationResult = {
  session_token: string;
  score_weights: Record<string, number>;
  recommendations: Recommendation[];
  disclaimer_en: string;
  disclaimer_zh: string;
};

File: backend/app/schemas/quiz.py
import uuid
from typing import Literal

from pydantic import BaseModel, Field


class QuizSessionCreate(BaseModel):
    language: Literal["en", "zh"] = "en"


class QuizSessionResponse(BaseModel):
    session_token: str
    language: str
    status: str


class QuizQuestionOption(BaseModel):
    code: str
    label_en: str
    label_zh: str


class QuizQuestionResponse(BaseModel):
    code: str
    question_en: str
    question_zh: str
    answer_type: Literal["single", "multiple"]
    options: list[QuizQuestionOption]


class QuizAnswersRequest(BaseModel):
    answers: dict[str, str | list[str]] = Field(min_length=1)


class RecommendationItem(BaseModel):
    product_id: uuid.UUID
    product_code: str
    product_name_en: str
    product_name_zh: str | None
    score: int
    tier: Literal["Excellent", "Good", "Potential", "Do Not Recommend"]
    reasons: list[str]


class RecommendationResponse(BaseModel):
    session_token: str
    score_weights: dict[str, int]
    recommendations: list[RecommendationItem]
    disclaimer_en: str
    disclaimer_zh: str

